cli/kaos_cli/facades/tests/test_backend_facade.py

```diff
@@ -0,0 +1,368 @@
+from parameterized import parameterized
+import re
+from unittest import TestCase
+
+from kaos_cli.facades.backend_facade import BackendFacade
+from kaos_cli.services.terraform_service import TerraformService
+from kaos_cli.services.state_service import StateService
+from kaos_cli.services.terraform_service import Command
+from kaos_cli.constants import CONFIG_PATH, ACTIVE, DEFAULT, CONTEXTS
+
+
+def backend_facade_test_inputs():
+    return [
+        (
+            'DOCKER', 'backend', 'url', 'token', 'infrastructure', 'kubeconfig', 'user'
+        ),
+        (
+            'GCP', 'backend', 'url', 'token', 'infrastructure', 'kubeconfig', 'user'
+        ),
+        (
+            'AWS', 'backend', 'url', 'token', 'infrastructure', 'kubeconfig', 'user'
+        ),
+        (
+            'Remote1', 'backend', 'url', 'token', 'infrastructure', 'kubeconfig', 'user'
+        ),
+    ]
+
+
+class TestBackendFacade(TestCase):
+
+    @staticmethod
+    def derive_context(cloud):
+        env_list = ['', 'dev', 'stage', 'prod']
+        for env in env_list:
+            return cloud + '_' + env
+
+    def manipulate_context_list(self, context_list):
+        self.state_service.set(CONTEXTS, environments=context_list)
+
+    # Arrange
+    TestCase.state_service = StateService()
+    TestCase.command = Command()
+    TestCase.tf_service = TerraformService(cmd=TestCase.command)
```

Comment:
This is an anti-pattern. Please avoid maintaining state between unit tests as you won't be able to isolated the causes and effects that are testing for properly. Instead, please create an instance of the `StateService` in each unit test and carry out the mutations you're testing for inside that test. The test should have no side-effect, neither local nor global, neither inside nor outside the method or the test case.

Reply:
Yes, this makes sense. Now I have moved the instantiation and initialization in the same ´´´arrange_test_fixtures()´´ method block, ensuring that the same ´´´StateService´´´state is not used across tests.